admiral/envs/components/health.py

```diff
@@ -121,19 +70,56 @@ def __init__(self, agents=None, entropy=0.1, **kwargs):
         self.agents = agents
         self.entropy = entropy
     
-    def reset(self, **kwargs):
+    def reset(self, agent, **kwargs):
```

Comment:
Per-agent reset is annoying and inconsistent with the design. A state reset is the reset of an environment's state, and all the agents in that environment are involved. Update this in all state components.